electrumx/server/db.py

```diff
@@ -1196,6 +1213,24 @@ def get_atomical_id_by_atomical_number(self, atomical_number):
             return None
         return atomical_id_value
     
+    # get rawtx by txnum
+    def get_rawtx_by_tx_hash(self, tx_hash):
+        tx_key = b'rtx' + tx_hash
+        rawtx = self.utxo_db.get(tx_key)
+        if not rawtx:
+            self.logger.error(f'get_rawtx {hash_to_hex_str(tx_hash)} txid not found')
+            return None
+        return rawtx
+    
+    def get_op_by_tx_num(self, tx_num):
+        op_key = b'op' + pack_le_uint64(tx_num)
+        op_data = self.utxo_db.get(op_key)
+        if not op_data:
+            self.logger.error(f'get_op {tx_num} txid not found')
+            return 9
```

Comment:
Return 9 is intended? Should it throw an exception instead?

Reply:
No need, return None will better